lit_stablelm/tokenizer.py

```diff
@@ -25,8 +25,26 @@ def token_to_id(self, token: str) -> int:
             raise ValueError(f"token {token!r} not found in the collection.")
         return id_
 
-    def encode(self, string: str, device: Optional[torch.device] = None) -> torch.Tensor:
+    def encode(
+        self, 
+        string: str, 
+        device: Optional[torch.device] = None, 
+        bos: bool = False,
```

Comment:
Note: I set bos=False as default here to make the existing tokenizer test pass in this repo. This is different from the Lit-LLaMA version. 

@carmocca @lantiga Let me know if this is undesired, we can also adapt the test.

Reply:
This is fine, I believe GPT only has the padding token

Reply:
Do we need the token arguments at all? Isn't this managed by the HFTokenizer internally?

BTW the changes here should be reflected in `test_tokenizer.py`